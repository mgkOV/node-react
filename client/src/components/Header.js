import '../styles/header.css';
import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import Payments from './Payments';

class Header extends Component {
  renderContent() {
    switch (this.props.auth) {
      case null:
        return <div className="loader">Loading...</div>;
      case false:
        return (
          <li>
            <a href="/auth/google">Login with Google</a>
          </li>
        );
      default:
        return (
          <li>
            <a href="/api/logout">Logout</a>
          </li>
        );
    }
  }

  render() {
    return (
      <nav>
        <div className="nav-wrapper">
          <Link
            to={this.props.auth ? '/surveys' : '/'}
            href="#"
            className="brand-logo"
          >
            Logo
          </Link>
          <ul className="right">
            {this.renderContent()}
          </ul>
        </div>
      </nav>
    );
  }
}

const mapStateToProps = ({ auth }) => ({
  auth
});

export default connect(mapStateToProps)(Header);
