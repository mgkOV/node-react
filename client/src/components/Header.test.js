import React from 'react';
import { shallow, render } from 'enzyme';
import { Provider } from 'redux';
import store from '../store';
import Header, { Unwrapped as UnwrappedHeader } from './Header';

describe('Header tests', () => {
  test('Header renders correctly', () => {
    const component = shallow(<UnwrappedHeader auth={{ credit: 5 }} />);
    expect(component).toMatchSnapshot();
  });
});
