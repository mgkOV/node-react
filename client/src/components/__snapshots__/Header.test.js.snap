// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header tests Header renders correctly 1`] = `
<nav>
  <div
    className="nav-wrapper"
  >
    <Link
      className="brand-logo"
      href="#"
      replace={false}
      to="/surveys"
    >
      Logo
    </Link>
    <ul
      className="right"
    >
      <li>
        <Connect(Payments) />
      </li>
      <li
        style={
          Object {
            "margin": "0 10px",
          }
        }
      >
        Credits: 
      </li>
      <li>
        <a
          href="/api/logout"
        >
          Logout
        </a>
      </li>
    </ul>
  </div>
</nav>
`;
